ThunderCatsTests/UIImageTests.m

```diff
@@ -0,0 +1,497 @@
+
+    XCTAssertEqual(fitSize.height, image.size.width);
+    XCTAssertEqual(fitSize.height, image.size.height);
+    
+    CIImage *ciImage = [[CIImage alloc] init];
+    UIImage *emptyImage = [[UIImage alloc] initWithCIImage:ciImage];
+    
+    XCTAssertNil([UIImage tc_imageWithImage:emptyImage scaledToFillSize:fitSize]);
+}
+
+- (void)testImageScaledToFill {
+    NSString *thePath = [[NSBundle bundleForClass:[self class]] pathForResource:@"icon-green" ofType:@"png"];
+    NSData *data = [[NSData alloc] initWithContentsOfFile:thePath];
+    UIImage *sampleImage = [[UIImage alloc] initWithData:data];
+    
+    CGSize fillSize = CGSizeMake(5.0, 10.0);
+    
+    UIImage *image = [UIImage tc_imageWithImage:sampleImage scaledToFitSize:fillSize];
+    
+    XCTAssertEqual(5.0, image.size.width);
+    XCTAssertEqual(5.0, image.size.height);
+    
+    CIImage *ciImage = [[CIImage alloc] init];
+    UIImage *emptyImage = [[UIImage alloc] initWithCIImage:ciImage];
+    
+    XCTAssertNil([UIImage tc_imageWithImage:emptyImage scaledToFitSize:CGSizeMake(5.0, 10.0)]);
+}
+
+- (void)testImageScaledToCropAndFill {
+    NSString *thePath = [[NSBundle bundleForClass:[self class]] pathForResource:@"icon-green" ofType:@"png"];
+    NSData *data = [[NSData alloc] initWithContentsOfFile:thePath];
+    UIImage *sampleImage = [[UIImage alloc] initWithData:data];
+    
+    UIImage *image = [UIImage tc_imageWithImage:sampleImage scaledAndCroppedToFillSize:CGSizeMake(5.0, 10.0)];
+    
+    XCTAssertEqual(5.0, image.size.width);
+    XCTAssertEqual(10.0, image.size.height);
+    
+    CIImage *ciImage = [[CIImage alloc] init];
+    UIImage *emptyImage = [[UIImage alloc] initWithCIImage:ciImage];
+    
+    XCTAssertNil([UIImage tc_imageWithImage:emptyImage scaledAndCroppedToFillSize:CGSizeMake(5.0, 10.0)]);
+}
+
+- (void)testALAssetOrientationConversion {
+    
+    XCTAssertEqual(UIImageOrientationDown, [UIImage tc_imageOrientationFromAssetOrientation:ALAssetOrientationDown]);
+    XCTAssertEqual(UIImageOrientationDownMirrored, [UIImage tc_imageOrientationFromAssetOrientation:ALAssetOrientationDownMirrored]);
+    XCTAssertEqual(UIImageOrientationUp, [UIImage tc_imageOrientationFromAssetOrientation:ALAssetOrientationUp]);
+    XCTAssertEqual(UIImageOrientationUpMirrored, [UIImage tc_imageOrientationFromAssetOrientation:ALAssetOrientationUpMirrored]);
+    XCTAssertEqual(UIImageOrientationLeft, [UIImage tc_imageOrientationFromAssetOrientation:ALAssetOrientationLeft]);
+    XCTAssertEqual(UIImageOrientationLeftMirrored, [UIImage tc_imageOrientationFromAssetOrientation:ALAssetOrientationLeftMirrored]);
+    XCTAssertEqual(UIImageOrientationRight, [UIImage tc_imageOrientationFromAssetOrientation:ALAssetOrientationRight]);
+    XCTAssertEqual(UIImageOrientationRightMirrored, [UIImage tc_imageOrientationFromAssetOrientation:ALAssetOrientationRightMirrored]);
+}
+
+- (void)testImageFromAsset {
```

Comment:
This method needs to be updated to use expectations.  http://nshipster.com/xctestcase/